pyciemss/interfaces.py

```diff
@@ -528,16 +533,204 @@ def wrapped_model():
+        bounds_interventions: List[List[float]]
+            - The lower and upper bounds for intervention parameter.
+            - Bounds are a list of the form [[lower bounds], [upper bounds]]
+        solver_method: str
+            - The method to use for solving the ODE. See torchdiffeq's `odeint` method for more details.
+            - If performance is incredibly slow, we suggest using `euler` to debug.
+              If using `euler` results in faster simulation, the issue is likely that the model is stiff.
+        solver_options: Dict[str, Any]
+            - Options to pass to the solver. See torchdiffeq' `odeint` method for more details.
+        start_time: float
+            - The start time of the model. This is used to align the `start_state` from the
+              AMR model with the simulation timepoints.
+            - By default we set the `start_time` to be 0.
+        inferred_parameters: Optional[pyro.nn.PyroModule]
+            - A Pyro module that contains the inferred parameters of the model.
+              This is typically the result of `calibrate`.
+            - If not provided, we will use the default values from the AMR model.
+        n_samples_ouu: int
+            - The number of samples to draw from the model to estimate risk for each optimization iteration.
+        maxiter: int
+            - Maximum number of basinhopping iterations: >0 leads to multi-start
+        maxfeval: int
+            - Maximum number of function evaluations for each local optimization step
+
+    Returns:
+        result: Dict[str, torch.Tensor]
+            - Dictionary of outputs from the model.
+                - Each key is the name of a parameter or state variable in the model.
+                - Each value is a tensor of shape (num_samples, num_timepoints) for state variables
+                    and (num_samples,) for parameters.
+    """
+    control_model = CompiledDynamics.load(model_path_or_json)
+    bounds_np = np.atleast_2d(bounds_interventions)
+    u_min = bounds_np[0, :]
+    u_max = bounds_np[1, :]
+    # Set up risk estimation
+    RISK = computeRisk(
+        model=control_model,
+        interventions=static_parameter_interventions,
+        qoi=qoi,
+        end_time=end_time,
+        logging_step_size=logging_step_size,
+        start_time=start_time,
+        risk_measure=lambda z: alpha_superquantile(z, alpha=0.95),
+        num_samples=1,
+        guide=inferred_parameters,
+        solver_method=solver_method,
+        solver_options=solver_options,
+    )
+
+    # Run one sample to estimate model evaluation time
+    start_t = time.time()
+    init_prediction = RISK.propagate_uncertainty(initial_guess_interventions)
+    RISK.qoi(init_prediction)
+    end_t = time.time()
+    forward_time = end_t - start_t
+    time_per_eval = forward_time / 1.0
+    if verbose:
+        print(f"Time taken: ({forward_time/1.:.2e} seconds per model evaluation).")
```

Comment:
In a future PR we'll want to change this to modify an external (TA4 provided) logger in-place. Not a concern now, just making a note.